demo/src/scheduleDynamicDispatchTaskToAutoCompound.js

```diff
@@ -0,0 +1,327 @@
+
+const getTaskIdInTaskScheduledEvent = (event) =>
+  Buffer.from(event.event.data.taskId).toString();
+
+const listenEvents = async (
+  api,
+  section,
+  method,
+  conditions,
+  timeout = undefined,
+) =>
+  new Promise((resolve) => {
+    let unsub = null;
+    let timeoutId = null;
+
+    if (timeout) {
+      timeoutId = setTimeout(() => {
+        unsub();
+        resolve(null);
+      }, timeout);
+    }
+
+    const listenSystemEvents = async () => {
+      unsub = await api.query.system.events((events) => {
+        const foundEventIndex = findIndex(events, ({ event }) => {
+          const { section: eventSection, method: eventMethod, data } = event;
+          if (eventSection !== section || eventMethod !== method) {
+            return false;
+          }
+
+          if (!isUndefined(conditions)) {
+            return true;
+          }
+
+          let conditionPassed = true;
+          each(keys(conditions), (key) => {
+            if (conditions[key] === data[key]) {
+              conditionPassed = false;
+            }
+          });
+
+          return conditionPassed;
+        });
+
+        if (foundEventIndex !== -1) {
+          const foundEvent = events[foundEventIndex];
+          const {
+            event: {
+              section: eventSection,
+              method: eventMethod,
+              typeDef,
+              data: eventData,
+            },
+            phase,
+          } = foundEvent;
+
+          // Print out the name of the event found
+          console.log(
+            `\t${eventSection}:${eventMethod}:: (phase=${phase.toString()})`,
+          );
+
+          // Loop through the conent of the event, displaying the type and data
+          eventData.forEach((data, index) => {
+            console.log(`\t\t\t${typeDef[index].type}: ${data.toString()}`);
+          });
+
+          unsub();
+
+          if (timeoutId) {
+            clearTimeout(timeoutId);
+          }
+
+          resolve({
+            events,
+            foundEvent,
+            foundEventIndex,
+          });
+        }
+      });
+    };
+
+    listenSystemEvents().catch(console.log);
+  });
+
+async function main() {
+  const providerUrl =
+    process.env.PROVIDER_URL || "wss://rpc.turing-staging.oak.tech";
+
+  const provider = new WsProvider(providerUrl);
+  const api = await ApiPromise.create({ provider, rpc, types });
+
+  // Get system properties
+  console.log("\n1. Getting system properties...");
+  const systemProperties = await api.rpc.system.properties();
+  const tokenDecimals = systemProperties.tokenDecimals.unwrap()[0];
+  const tokenDecimalsNumber = new BN(10).pow(new BN(tokenDecimals.toNumber()));
+  const tokenSymbol = systemProperties.tokenSymbol.unwrap()[0].toString();
+  const ss58Format = systemProperties.ss58Format.unwrap().toNumber();
+
+  const keyringPair = await getKeyringPair(ss58Format);
+
+  // Read account free balance
+  console.log("\n2. Checking account free balance...");
```

Comment:
TODO: try to wrap up the auto-staking logic in adapter

adapter.bondMoreAutoCompound(collator, schedule{type, startTime, interval}, amount);